opensearch-operator/pkg/helpers/reconcile-helpers.go

```diff
@@ -78,8 +78,13 @@ func ResolveDashboardsImage(cr *opsterv1.OpenSearchCluster) (result opsterv1.Ima
 		defaultRepo = *cr.Spec.General.DefaultRepo
 	}
 
+	version := cr.Spec.Dashboards.Version
+	if version == "" {
+		version = cr.Spec.General.Version
```

Comment:
This will be a problem in scenarios where we only release OpenSearch. Example for 3.3.1 https://github.com/opensearch-project/opensearch-build/issues/5784. The docker for OpenSearch has 3.3.1 https://hub.docker.com/r/opensearchproject/opensearch/tags, but not for OpenSearch Dashboards https://hub.docker.com/r/opensearchproject/opensearch-dashboards/tags. 
CC @getsaurabh02 @peterzhuamazon